src/imageBuild/buildToolsRegistry.ts

```diff
@@ -0,0 +1,94 @@
+import * as vscode from 'vscode';
+
+import { getPreferBuildah } from '../components/config/config';
+import { shell } from '../shell';
+
+interface ContainerImageBuildTool {

```

Comment:
I like the way you've done this - should make it easy if people want to add other build tools in future.